internal/receiver/discoveryreceiver/README.md

```diff
@@ -0,0 +1,334 @@
+
+```
+2022-07-27T17:11:27.271Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "logging", "#logs": 1}
+2022-07-27T17:11:27.271Z	info	ResourceLog #0
+Resource SchemaURL:
+Resource labels:
+     -> event.type: STRING(statement.match)
+ScopeLogs #0
+ScopeLogs SchemaURL:
+InstrumentationScope
+LogRecord #0
+ObservedTimestamp: 2022-07-27 17:11:27.27149884 +0000 UTC
+Timestamp: 2022-07-27 17:11:27.271444488 +0000 UTC
+Severity: warn
+Body: Container appears to be accepting redis connections but the default auth is incorrect.
+Attributes:
+     -> runnerPID: STRING(119386)
+     -> monitorID: STRING(smartagentredisreceiver_creatordiscoveryendpoint1721702637958d3f3335daf83f7)
+     -> monitorType: STRING(collectd/redis)
+     -> caller: STRING(signalfx/handler.go:189)
+     -> name: STRING(smartagent/redis/receiver_creator/discovery{endpoint="172.17.0.2:6379"}(58d3f3335daf83f7))
+     -> createdTime: STRING(1.658941887269115e+09)
+     -> lineno: STRING(201)
+     -> logger: STRING(root)
+     -> sourcePath: STRING(/usr/lib/splunk-otel-collector/agent-bundle/collectd-python/redis/redis_info.py)
+     -> level: STRING(error)
+     -> receiver.name: STRING(smartagent/redis)
+     -> discovery.status: STRING(partial)
+Trace ID:
+Span ID:
+Flags: 0
+```
+
+If the Docker observer reports an unrelated container that isn't running Redis, the following log record would be emitted:
+
+```
+2022-07-27T17:16:57.718Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "logging", "#logs": 1}
+2022-07-27T17:16:57.719Z	info	ResourceLog #0
+Resource SchemaURL:
+Resource labels:
+     -> event.type: STRING(statement.match)
+ScopeLogs #0
+ScopeLogs SchemaURL:
+InstrumentationScope
+LogRecord #0
+ObservedTimestamp: 2022-07-27 17:16:57.718720008 +0000 UTC
+Timestamp: 2022-07-27 17:16:57.718670678 +0000 UTC
+Severity: debug
+Body: Container appears to not be accepting redis connections.
+Attributes:
+     -> name: STRING(smartagent/redis/receiver_creator/discovery{endpoint="172.17.0.2:80"}(54141aa1da2d2ad0))
+     -> monitorType: STRING(collectd/redis)
+     -> logger: STRING(root)
+     -> createdTime: STRING(1.6589422177182763e+09)
+     -> level: STRING(error)
+     -> monitorID: STRING(smartagentredisreceiver_creatordiscoveryendpoint17217028054141aa1da2d2ad0)
+     -> runnerPID: STRING(119590)
+     -> sourcePath: STRING(/usr/lib/splunk-otel-collector/agent-bundle/collectd-python/redis/redis_info.py)
+     -> lineno: STRING(198)
+     -> caller: STRING(signalfx/handler.go:189)
+     -> receiver.name: STRING(smartagent/redis)
+     -> discovery.status: STRING(failed)
+Trace ID:
+Span ID:
+Flags: 0
+```
+
+## Config
+
+### Main
+
+| Name | Type | Default | Docs |
+| ---- | ---- | ------- | ---- |
+| `watch_observers` (required) | []string | <no value> | The array of Observer extensions to receive Endpoint events from |
+| `log_endpoints` | bool | false | Whether to emit log records for Observer Endpoint events |
```

Comment:
Correct me if I am wrong, with the log_endpoints option, a new receiver type (eg.  observer_events receiver) would not be needed. Does that mean we are following the inventory events [representation](https://docs.google.com/document/d/17gc7i9_zRJn_ztKNbk-jvyXMhiXXn4tjvmojW0NNf0U/edit#) to help make it easier to integrate with the inventory service once it is available?

Reply:
Right, since this receiver provides an observer.Notify implementation and emits log records it's straightforward for it to also log all endpoint events from configured observers. My thought was to emit log records that contain endpoints themselves since they're an established model in the collector: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/observer/endpoints.go (content documented here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md#rule-expressions). We could provide conversion in the signalfx/hec exporters if their form isn't suitable for backend requirements.